src/basic/strv.h

```diff
@@ -96,10 +96,21 @@ bool strv_overlap(char **a, char **b) _pure_;
 #define STRV_FOREACH(s, l)                      \
         for ((s) = (l); (s) && *(s); (s)++)
 
-#define STRV_FOREACH_BACKWARDS(s, l)            \
-        STRV_FOREACH(s, l)                      \
-                ;                               \
-        for ((s)--; (l) && ((s) >= (l)); (s)--)
+#define STRV_FOREACH_BACKWARDS(s, l)                                \
+        for (s = ({                                                 \
+                        char **_l = l;                              \
+                        if (l) {                                    \
+                                unsigned _c;                        \
+                                _c = strv_length(_l);               \
+                                if (_c <= 0)                        \
```

Comment:
actually, nope, the old code would return l-1 if l was empty but non-NULL. Your expression would return l in that case, which is not intended.

That said, we can handle this nicer in a different way. Will rework this.